api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -257,6 +257,10 @@ public void deleteWorkspace(DbWorkspace dbWorkspace) {
     try {
       fireCloudService.deleteBillingProject(billingProjectName);
       billingProjectAuditor.fireDeleteAction(billingProjectName);
+      // Since deleted workspace entry still exist in database we have to explicitly remove it from
+      // featured_workspace
+      // if they exist
+      featuredWorkspaceDao.deleteDbFeaturedWorkspaceByWorkspace(dbWorkspace);
```

Comment:
the Terra above can be fail for various reason, we need to move this up to close to workspaceDao..